[Remove Vuetify from Studio] Delete, restore, and visibility confirmation dialogs in admin channel actions

Prashant-thakur77 opened the pull request:
Fixes #5410 

## Summary
Migrated 5 channel action confirmation dialogs from Vuetify to KDS KModal

Changed title of the soft delete button from"Permanently delete channel" to "delete channel" for proper distinction (after confirming from misrob)

## After images for Desktop users:

<img width="1347" height="685" alt="Screenshot From 2025-10-01 17-33-08" src="https://github.com/user-attachments/assets/fda304e5-68a3-46e4-8720-d8ba9d1154a6" />
<img width="1347" height="675" alt="Screenshot From 2025-10-01 17-33-20" src="https://github.com/user-attachments/assets/aad3160b-2534-4696-ae18-3b617dd79841" />
<img width="1347" height="675" alt="Screenshot From 2025-10-01 17-33-25" src="https://github.com/user-attachments/assets/62ce32c7-0009-4b39-a3d8-066b6b61e40f" />
<img width="1347" height="675" alt="Screenshot From 2025-10-01 17-33-28" src="https://github.com/user-attachments/assets/e7064eb0-e730-440e-ae02-4852b123dec5" />
<img width="1347" height="675" alt="Screenshot From 2025-10-01 17-33-31" src="https://github.com/user-attachments/assets/44e2eb4e-dda4-4c7c-92c5-f6ffcbb61cb0" />





## After images for Mobile users:

<img width="399" height="685" alt="Screenshot From 2025-10-01 17-17-06" src="https://github.com/user-attachments/assets/a05f206d-e909-46cc-b46e-1b5cd38720fa" />
<img width="399" height="685" alt="Screenshot From 2025-10-01 17-17-16" src="https://github.com/user-attachments/assets/e849a8b0-0469-4a89-9428-1d77f5935e35" />
<img width="399" height="685" alt="Screenshot From 2025-10-01 17-17-24" src="https://github.com/user-attachments/assets/17f8173c-6298-4b76-893b-530c25e66f38" />
<img width="399" height="685" alt="Screenshot From 2025-10-01 17-18-05" src="https://github.com/user-attachments/assets/06d9efd6-88b2-4a7d-a6aa-003d9eb1d196" />






## References
• Parent issue: https://github.com/learningequality/studio/issues/5060
## Reviewer guidance
Login as a@a.com with password a
Go to Administration > Channels
Click Actions dropdown in the last column of the table
Click Make public / Make private / Delete channel / Restore / Delete pernamently (note that availability of these options depends on channel type).

Prashant-thakur77 commented:
Hi @MisRob,

I've updated the ChannelActionsDropdown.spec.js tests (Locally) after migrating to KModal as i saw the tests failing.

Changes:
    Updated event names from 'confirm' to 'submit' (KModal uses different events).
    Added dialog activation steps (KModal uses v-if and needs to be rendered first).
    All tests now pass locally.


Let me know if you have any feedback.

MisRob commented:
Thank you @Prashant-thakur77, we will assign a reviewer next week.

MisRob commented:
So you can just remove the above part from the code :)

Generally tokens can be only applied via computed style or class as [described here](https://design-system.learningequality.org/colors#usage). But again, here I think you won't need it.

Prashant-thakur77 commented:
@MisRob The issue I’m facing is that when the Delete Channel action is triggered, the text across the entire table turns red. Since the modal is a child of this parent component, the text within the KModal — both the title and content — also inherits the red color when the modal is restored or when Permanently Delete is clicked.

I’ve managed to fix the content color issue by applying inline styles directly to the div, but I’m uncertain about the correct way to set the title text color.

Additionally, since the table components are center-aligned by default, applying a left alignment to the modal text and title seems important — otherwise, both elements appear centered within the modal.

Prashant-thakur77 commented:
these are without the styles:

<img width="480" height="270" alt="Screenshot From 2025-10-06 21-23-49" src="https://github.com/user-attachments/assets/a5fea9df-4428-443d-bce7-9f1c28929f0b" />
<img width="480" height="270" alt="Screenshot From 2025-10-06 21-23-58" src="https://github.com/user-attachments/assets/f370ef2f-102a-4e76-ba2c-14d8fbe47d33" />

MisRob commented:
That's helpful @Prashant-thakur77, thank you.

It'd be best to fix this right in `KModal`. It likely inherits these Studio styles because there are no color, alignment, and weight styles set directly on the title from within the modal. Would you be interested in resolving it in KDS repository as part of this work?

MisRob commented:
Thank you @Prashant-thakur77. So the first step would be to try if this suggestion helps with this Studio use-case.

These is [development documentation for KDS](https://github.com/learningequality/kolibri-design-system/tree/develop/dev_docs). Note that information on linking is obsolete in the dev docs. You can link your local KDS to local Studio with running these commands in Studio: `pnpm install file:../kolibri-design-system` and then `pnpm run devserver`. Note that when you make any changes in KDS repository, you will need to stop your Studio devserver and re-run those two commands again to changes to propagate. Also, this may not work for everyone. In case you experienced any problems, let us know.

Prashant-thakur77 commented:
@MisRob 
Just wanted to let you know that I’ve completed my setup and everything is working perfectly. Thank you for your valuable tips — they were a great help!

Prashant-thakur77 commented on contentcuration/contentcuration/frontend/administration/pages/Channels/ChannelActionsDropdown.vue:
```diff
@@ -126,37 +126,37 @@
-            title: this.$tr('makePrivateTitle'),
-            submitText: this.$tr('makePrivateAction'),
-            message: this.$tr('makePrivateMessage', { name: this.name }),
+            title: 'Make channel private',
```
Following the feedback on this comment https://github.com/learningequality/studio/pull/5426#discussion_r2461098244
, i was advised to use direct strings in the administration section, and i18n is not required here

AllanOXDi reviewed (approved):
LGTM! Thanks @Prashant-thakur77 , @MisRob  will merge it

Prashant-thakur77 commented:
Hello @MisRob, regarding the resolution of conflicts:
>:error-text="communityChannelErrorMessage" 

and 
>:disable-submit="isCommunityChannel"

 were added to the MakePublicDialog since it was updated in the confirmation dialog.
Now, as I’m using a single KModal for all five dialogs, should I just pass the props like this?
>:errorMessage="dialogConfig.errorMessage"
:submitDisabled="dialogConfig.submitDisabled"

And then update the dialog config like this:
> makePublic: {
  title: this.$tr('makePublicTitle'),
  submitText: this.$tr('makePublicAction'),
  message: this.$tr('makePublicMessage', { name: this.name }),
  testId: 'confirm-public',
  handler: this.makePublicHandler,

Added this 
 >  errorMessage: this.communityChannelErrorMessage,   //Added this 
  submitDisabled: this.isCommunityChannel,       // and this
},

For MakePublic, and keep the rest of the four dialogs as:
> errorMessage: '',
submitDisabled: false,

MisRob commented:
@Prashant-thakur77, I am not sure if I follow the details, but overall yes you can just keep the same approach - `KModal` with the `dialogConfig`. And then when creating the config, just add some additional logic that reflect the new community channel logic. It'd best to preview the new community related channel dialog on the current `unstable` as is now, understand what it does, and then make sure that after you have rebased, that experience is preserved. This way, all should be good.

MisRob commented:
Yes this is ideal @Prashant-thakur77 - that was what I was going to suggest too.

MisRob commented on contentcuration/contentcuration/frontend/administration/pages/Channels/__tests__/channelActionsDropdown.spec.js:
Hi @Prashant-thakur77, generally yes, we don't want to test in Studio what is KDS's responsibility (for example that it properly renders buttons and such) Test only Studio-specific logic such as:

- (1) Dialog with correct content is displayed when the corresponding dropdown item is clicked
- (2) When the dialog's submit button is clicked, that the correct action (mock) is called
- (3) It's fine to test that the dialog disappears when submit clicked since you control it from Studio, but (2) is more important to cover.

MisRob replied:
Oh thanks - I'm glad my babbling about tests is not annoying, and sure that your work will be appreciated by developers who will work on these files in the future :) We have trouble with the approaches I mentioned in the whole codebase so hopefully we can gradually improve it.

Prashant-thakur77 commented:
i @MisRob  I've updated the test file with the following additions:
 Added comprehensive tests for mockActions to verify all expected functions are called with correct parameters
Please review when you have a moment and let me know if any adjustments is needed.

MisRob commented on contentcuration/contentcuration/frontend/administration/pages/Channels/ChannelActionsDropdown.vue:
```diff
@@ -1,56 +1,33 @@
+      @cancel="activeDialog = null"
+    >
+      <p>{{ dialogConfig.message }}</p>
+      <div
```
Yes

MisRob commented on contentcuration/contentcuration/frontend/administration/pages/Channels/__tests__/channelActionsDropdown.spec.js:
```diff
@@ -19,125 +22,299 @@ const channel = {
-  describe('public channel actions', () => {
-    beforeEach(() => {
-      [wrapper, mocks] = makeWrapper();
+  describe('menu visibility', () => {
```
Total nitpick - it may make sense to have this section as first in the test suite - then when it's read from up to bottom or in terminal log, it's a bit easier to process what a component does.

Prashant-thakur77 replied:
rightly said ,will update

Prashant-thakur77 replied:
Addressed it

Prashant-thakur77 commented on contentcuration/contentcuration/frontend/administration/pages/Channels/__tests__/channelActionsDropdown.spec.js:
```diff
@@ -19,125 +22,299 @@ const channel = {
+      const confirmButton = screen.getByRole('button', { name: 'Restore' });
+      await user.click(confirmButton);
+
+      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
```
sure,will update right now

Prashant-thakur77 replied:
Updated
